### PR TITLE
Enable Via for the Sofle

### PR DESCRIPTION
**Pull request by Sturek**
## Description
I kept as much of the original default keymap as possible. To make it more obvious which features cannot be changed in Via, I broke the OLED screens and the rotation of the encoders out into seperate files.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sofle/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/sofle/keymaps/via/config.h:1]: Could add a license header here, please?

Here's a sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/sofle/keymaps/via/encoder.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
+//Setting up what encoder rotation does. If your encoder can be pressed as a button, that function can be set in Via.
```
Comment (noroadsleft) [inline keyboards/sofle/keymaps/via/encoder.c:1]: Could add a license header here, please?

File: keyboards/sofle/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/sofle/keymaps/via/keymap.c:1]: Could add a license header here, please?

File: keyboards/sofle/keymaps/via/oled.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
+//Sets up what the OLED screens display.
```
Comment (noroadsleft) [inline keyboards/sofle/keymaps/via/oled.c:1]: Could add a license header here, please?

File: keyboards/sofle/keymaps/via/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+![SofleKeyboard default keymap](https://github.com/josefadamcik/SofleKeyboard/raw/master/Images/soflekeyboard.png)
+![SofleKeyboard adjust layer](https://github.com/josefadamcik/SofleKeyboard/raw/master/Images/soflekeyboard_layout_adjust.png)
+
+
+# Default keymap for Sofle Keyboard using Via
```
Comment (noroadsleft) [inline keyboards/sofle/keymaps/via/readme.md:5]:
```suggestion
# VIA keymap for Sofle
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/sofle/keymaps/via/config.h (hunk 1 above)
Comment (Sturek) [reply keyboards/sofle/keymaps/via/config.h:18]: I added the information to all files. Very sorry I did not have that right away, the default map does not and as I did so little modification to it, I was unsure if it was my place to put a license (and my name) on it. I hope my solution is okay. Thanks for your work!

**Comment by Sturek** [comment]
I should probably have commented here, sorry, still new to github. I'll paste my comment here for clarity.

I added the information to all files. Very sorry I did not have that right away, the default map does not and as I did so little modification to it, I was unsure if it was my place to put a license (and my name) on it. I hope my solution is okay. Thanks for your work!

**Comment by noroadsleft** [comment]
> I should probably have commented here, sorry, still new to github. I'll paste my comment here for clarity.
> 
> I added the information to all files. Very sorry I did not have that right away, the default map does not and as I did so little modification to it, I was unsure if it was my place to put a license (and my name) on it. I hope my solution is okay. Thanks for your work!

That's fine. The license headers are a fairly new requirement, and if you'll look around you'll likely find many files that should have them don't.

**Review by noroadsleft — APPROVED** [review]

File: keyboards/sofle/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+
+[0] = LAYOUT( \
+  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV, \
+  KC_ESC,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC, \
+  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT, \
+  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     XXXXXXX,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT, \
+                 KC_LGUI,KC_LALT,KC_LCTRL, MO(2), KC_ENT,      KC_SPC,  MO(3), KC_RCTRL, KC_RALT, KC_RGUI \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/via/keymap.c:46]:
```suggestion
[0] = LAYOUT(
  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV,
  KC_ESC,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,  KC_BSPC,
  KC_TAB,   KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L, KC_SCLN,  KC_QUOT,
  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,     XXXXXXX,KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT,
                 KC_LGUI,KC_LALT,KC_LCTRL, MO(2), KC_ENT,      KC_SPC,  MO(3), KC_RCTRL, KC_RALT, KC_RGUI
```

File: keyboards/sofle/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+
+[1] = LAYOUT( \
+  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV, \
+  KC_ESC,   KC_Q,   KC_W,    KC_F,    KC_P,    KC_G,                      KC_J,    KC_L,    KC_U,    KC_Y, KC_SCLN,  KC_BSPC, \
+  KC_TAB,   KC_A,   KC_R,    KC_S,    KC_T,    KC_D,                      KC_H,    KC_N,    KC_E,    KC_I,    KC_O,  KC_QUOT, \
+  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,      XXXXXXX,KC_K,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT, \
+                 KC_LGUI,KC_LALT,KC_LCTRL,KC_TRNS, KC_ENT,      KC_SPC,  KC_TRNS, KC_RCTRL, KC_RALT, KC_RGUI \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/via/keymap.c:69]:
```suggestion
[1] = LAYOUT(
  KC_GRV,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_GRV,
  KC_ESC,   KC_Q,   KC_W,    KC_F,    KC_P,    KC_G,                      KC_J,    KC_L,    KC_U,    KC_Y, KC_SCLN,  KC_BSPC,
  KC_TAB,   KC_A,   KC_R,    KC_S,    KC_T,    KC_D,                      KC_H,    KC_N,    KC_E,    KC_I,    KC_O,  KC_QUOT,
  KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_MUTE,      XXXXXXX,KC_K,    KC_M, KC_COMM,  KC_DOT, KC_SLSH,  KC_RSFT,
                 KC_LGUI,KC_LALT,KC_LCTRL,KC_TRNS, KC_ENT,      KC_SPC,  KC_TRNS, KC_RCTRL, KC_RALT, KC_RGUI
```

File: keyboards/sofle/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[2] = LAYOUT( \
+  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,\
+  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                       KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12, \
+  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE, \
+  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______, \
+                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______\
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/via/keymap.c:90]:
```suggestion
[2] = LAYOUT(
  _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,                       KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,
  KC_GRV,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,                       KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_F12,
  _______, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC,                       KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_PIPE,
  _______,  KC_EQL, KC_MINS, KC_PLUS, KC_LCBR, KC_RCBR, _______,       _______, KC_LBRC, KC_RBRC, KC_SCLN, KC_COLN, KC_BSLS, _______,
                       _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

File: keyboards/sofle/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+ *            | LGUI | LAlt | LCTR |LOWER | /Enter  /       \Space \  |RAISE | RCTR | RAlt | RGUI |
+ *            |      |      |      |      |/       /         \      \ |      |      |      |      |
+ *            `----------------------------------'           '------''---------------------------'
+ */
+[3] = LAYOUT( \
+  _______, _______ , _______ , _______ , _______ , _______,                           _______,  _______  , _______,  _______ ,  _______ ,_______, \
+  _______,  KC_INS,  KC_PSCR,   KC_APP,  XXXXXXX, XXXXXXX,                        KC_PGUP, _______,   KC_UP, _______,_______, KC_BSPC, \
+  _______, KC_LALT,  KC_LCTL,  KC_LSFT,  XXXXXXX, KC_CAPS,                       KC_PGDN,  KC_LEFT, KC_DOWN, KC_RGHT,  KC_DEL, KC_BSPC, \
+  _______,KC_UNDO, KC_CUT, KC_COPY, KC_PASTE, XXXXXXX,  _______,       _______,  XXXXXXX, _______, XXXXXXX, _______,   XXXXXXX, _______, \
+                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______ \
```
Comment (fauxpark) [inline keyboards/sofle/keymaps/via/keymap.c:111]:
```suggestion
[3] = LAYOUT(
  _______, _______ , _______ , _______ , _______ , _______,                           _______,  _______  , _______,  _______ ,  _______ ,_______,
  _______,  KC_INS,  KC_PSCR,   KC_APP,  XXXXXXX, XXXXXXX,                        KC_PGUP, _______,   KC_UP, _______,_______, KC_BSPC,
  _______, KC_LALT,  KC_LCTL,  KC_LSFT,  XXXXXXX, KC_CAPS,                       KC_PGDN,  KC_LEFT, KC_DOWN, KC_RGHT,  KC_DEL, KC_BSPC,
  _______,KC_UNDO, KC_CUT, KC_COPY, KC_PASTE, XXXXXXX,  _______,       _______,  XXXXXXX, _______, XXXXXXX, _______,   XXXXXXX, _______,
                         _______, _______, _______, _______, _______,       _______, _______, _______, _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**